pumpkin/src/client/client_packet.rs

```diff
@@ -106,10 +106,23 @@ impl Client {
                 profile_actions: None,
             });
 
-            // TODO: check config for encryption
-            let verify_token: [u8; 4] = rand::random();
-            self.send_packet(&server.encryption_request(&verify_token, BASIC_CONFIG.online_mode))
+            if BASIC_CONFIG.encryption {
+                let verify_token: [u8; 4] = rand::random();
+                self.send_packet(
+                    &server.encryption_request(&verify_token, BASIC_CONFIG.online_mode),
+                )
                 .await;
+            } else {
+                if ADVANCED_CONFIG.packet_compression.enabled {
+                    self.enable_compression().await;
+                }
+                self.finish_login(
+                    gameprofile
+                        .as_ref()
+                        .expect("The GameProfile was just set to Some, this should never fail"),
+                )
```

Comment:
You could actually remove the need for an optional gameprofile here. Above a new gameprofile is creates, you can store it in a new var
```rust
            let profile = GameProfile {
                id: login_start.uuid,
                name: login_start.name,
                properties: vec![],
                profile_actions: None,
            };
            *gameprofile = Some(profile);
            
             self.finish_login(
                   profile,
                )
 ```

Reply:
borrow checker would not like this as we move profile into gameprofile

Reply:
`finish_login` takes in a reference, means you also can set the gameprofile below the function